wsdl/ver10/schema/metadatastream.xsd

```diff
@@ -389,6 +389,34 @@ IN NO EVENT WILL THE CORPORATION OR ITS MEMBERS OR THEIR AFFILIATES BE LIABLE FO
 		</xs:attribute>
 		<xs:anyAttribute processContents="lax"/>
 	</xs:complexType>
+	<xs:complexType name="ObjectSnapshot">
+		<xs:sequence>
+			<xs:element name="ImageRef" type="xs:anyURI" minOccurs="0"/>
+			<xs:element name="Image" type="xs:base64Binary" minOccurs="0"/>
+			<xs:element name="BoundingBox" type="tt:Rectangle" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Bounding box definition in normalized coordinates.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:any namespace="##any" processContents="lax" minOccurs="0" maxOccurs="unbounded"/>      <!-- first ONVIF then vendor --> 
+		</xs:sequence>
+		<xs:attribute name="CaptureTime" type="xs:dateTime"/>
+		<xs:anyAttribute processContents="lax"/>
+	</xs:complexType>
+	<xs:complexType name="ObjectTrackComplement">
```

Comment:
Since this "ObjectTrackComplement" complex type is not used  in metadata do we have to define this in metadatastream.xsd file ? do we have to define this in onvif.xsd or rule.xsd file?

Reply:
rules.xsd seems to be a special invention for motion region configuration and has not been touched ever since 19.06! Also since we are defining this as an event, guess the event can be put into the metadata stream as any other event and hence felt metatastream.xsd is an appropriate place especially when building on top of tt:Object with an additional capture time attribute.